src/webview-contract/webviewDefinitions/connectAcrToCluster.ts

```diff
@@ -0,0 +1,93 @@
+import { WebviewDefinition } from "../webviewTypes";
+
+export type SubscriptionKey = {
+    subscriptionId: string;
+};
+
+export type Subscription = SubscriptionKey & {
+    name: string;
+};
+
+export type AcrKey = SubscriptionKey & {
+    resourceGroup: string;
+    acrName: string;
+};
+
+export type Acr = AcrKey; // Fully-defined by its key
+
+export type ClusterKey = SubscriptionKey & {
+    resourceGroup: string;
+    clusterName: string;
+};
+
+export type Cluster = ClusterKey; // Fully-defined by its key
+
+// https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles
+export const acrPullRoleDefinitionName = "7f951dda-4ed3-4680-a7ca-43fe172d538d";
```

Comment:
I agree it feels a bit uncomfortable to rely on hard-coded GUID values. An alternative is to search for role definitions based on their text name (`AcrPull` in this case), but that seems more risky to me, especially as some role definitions have very non-identifier-sounding names like "Azure Kubernetes Service Cluster User Role".

I've verified that these role definition GUID names are consistent across tenants. They are hard-coded in thousands of places across the Azure organization in GitHub, and any changes would break existing customer role assignments, so it would not make any sense for them to be changed.